packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,40 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	private caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+		if (clientOptions.caseInsensitiveCommands) {
+			// Because slash commands must be lowercase anyways, blanketing all commands to lowercase is safe.
+			this.parsedSubcommandMappings.forEach((cmd) => cmd.name.toLowerCase());
+			this.caseInsensitiveSubCommands = true;
```

Comment:
```suggestion
			this.caseInsensitiveSubCommands = true;

			// Because slash commands must be lowercase anyway, we can transform all to lowercase.
			for (const cmd of this.parsedSubcommandMappings) {
				cmd.name = cmd.name.toLowerCase();
			}
```

![](https://play.pokemonshowdown.com/sprites/ani/ditto.gif)